packages/knip/src/plugins/vite/index.ts

```diff
@@ -12,6 +18,52 @@ const isEnabled: IsPluginEnabled = ({ dependencies }) => hasDependency(dependenc
 
 export const config = ['vite.config.{js,mjs,ts,cjs,mts,cts}'];
 
+const hasReactBabelPlugins = (config: ViteConfig) => {
+  if (!config?.plugins?.length) {
+    return false;
+  }
+
+  for (const plugin of config.plugins) {
+    if (typeof plugin === 'function') {
+      const pluginConfig = plugin() as BabelPlugin;
+      if (pluginConfig?.babel?.plugins) {
+        return true;
+      }
+    } else if (Array.isArray(plugin)) {
+      const [pluginObj] = plugin as [BabelPlugin];
+      if (pluginObj?.babel?.plugins) {
+        return true;
+      }
+    } else if (typeof plugin === 'object' && plugin !== null) {
+      const pluginObj = plugin as BabelPlugin;
+      if (pluginObj?.babel?.plugins) {
+        return true;
+      }
+    }
+  }
+
+  return false;
```

Comment:
I didn't realize at first the babel object is sent to the react plugin, but if this works then... all good! :)